onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc

```diff
@@ -212,18 +263,18 @@ bool IsPReluOpSupported(const Node& node, const OpBuilderInputParams& input_para
 bool ActivationOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderInputParams& input_params,
                                             const logging::Logger& logger) const {
   const auto& op_type = node.OpType();
+#if !defined(COREML_ENABLE_MLPROGRAM)
+  if (op_type == "Gelu") {
+    return false;
+  }
+#endif
 
-#if defined(COREML_ENABLE_MLPROGRAM)
-  if (input_params.create_mlprogram) {
-    if (op_type == "PRelu") {  // TODO: ML Program supports this so should be easy to enable
-      return false;
-    }
-  } else
-#endif  // (COREML_ENABLE_MLPROGRAM)
-  {
-    if (op_type == "PRelu") {
-      return IsPReluOpSupported(node, input_params, logger);
-    }
+  if (op_type == "Gelu" && !input_params.create_mlprogram) {
```

Comment:
nit: when `!defined(COREML_ENABLE_MLPROGRAM)`, there are two conditions that check `op_type == "Gelu"`. do we need the one in the ifdef above?